### DISCUSSION TITLE
What's the best approach for style and onClick the next/link component in 12.1.6?

### DISCUSSION THREAD

**Discussion by danielpl10** [vercel/next.js#36988] [Help]
There are situations where you want to style a link and also do other functionalities beside navigate. The approach for these two things that I'm always seeing is to have something like this:

```
<Link href="...">
     <a className="......" onClick={.....}>
         {title}
     </a>
</Link>
```

Is there a better approach to do this without having an `<a>` tag inside `<Link>` in the new version 12.1.6?

**Comment by icyJoseph** [comment]
I know that a Link from next/image works without nesting an `a` tag, but I always include it, that way I can keep some degree of separation between the Link and the actual component I use, of course, not forgetting to include the `passHref` prop, when abstracting the component.

I think that's where we are at right now. Sometimes, but not always, the additional functionality can be done in other ways, but I can understand it becomes slightly verbose to include an anchor tag and a callback all at once. At the end of the day, the Link exists, pretty much to tell the next/router that the user wants to navigate, like in any other router framework.
